xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -513,25 +513,60 @@ posix_do_fchmod(xlator_t *this, int fd, struct iatt *stbuf)
     return ret;
 }
 
+#if defined(HAVE_FUTIMENS)
+
+typedef struct timespec gf_futime_t;
```

Comment:
Why not declare `gf_utime_t` always as a `struct timespec` and only when `futimens()` is not supported, we convert it to timeval just before calling `futimes()` in `gf_futime()`.

I think this makes it more consistent and Gluster always works with nanosecond resolution. Only if/when a system call requires it, we truncate to microseconds.

Reply:
OK so we don't need `gf_utime_t` at all.